testapp/models.py

```diff
@@ -0,0 +1,58 @@
+from django.db import models
+from django.db.models import JSONField
+
+from django_mongodb.managers import MongoManager
+
+
+class FooModel(models.Model):
+    objects: MongoManager = MongoManager()
+
+    json_field = JSONField()
+    name = models.CharField(max_length=100)
+    datetime_field = models.DateTimeField(auto_now_add=True)
+    time_field = models.TimeField(auto_now_add=True)
+    date_field = models.DateField(auto_now_add=True)
+
+
+class SameTableChild(FooModel):
+    dummy_model_ptr = models.OneToOneField(
+        FooModel,
+        on_delete=models.CASCADE,
+        parent_link=True,
+        related_name="child_type",
+        db_column="_id",
+    )
+    extended = models.CharField(max_length=100)
+
+    class Meta:
+        db_table = "testapp_foomodel"
```

Comment:
This child model is persistent with it's and it's parents fields in the same collection.